lib/logstash/filters/kv.rb

```diff
@@ -411,8 +411,10 @@ def filter(event)
     return if kv.empty?
 
     if @target
-      @logger.debug? && @logger.debug("Overwriting existing target field", :target => @target)
-      event.set(@target, kv)
+      @logger.debug? && @logger.debug("Merging into existing target field", :target => @target)
+      t = event.get(@target)
+      t = {} unless t.is_a?(Hash)
```

Comment:
Consider that if `t` is a scalar or array, it just goes away, this becomes a "if target is a hash: merge, otherwise overwrite" operation.

I think this is a very uncommon edge case that can be covered by updating the docs to warn the user, that if they want to keep the value at `target` they should stash the value at `target` and add it back to the new `target` hash at the key of their choosing. If we support #11 then the `add back` will not fail because target is always a hash.

Reply:
Yes, that was my reasoning: if `t` is anything else than a hash then drop it. Thanks for mentioning #11 - this is also a legitimate concern indeed. Also looking at the `add_field` behaviour, it will turn the value into an array if the field exists, so we do try to avoid loosing values.